CMakeLists.txt

```diff
@@ -132,7 +132,7 @@ else()
 endif()
 set(MIOPEN_BINCACHE_PATH "" CACHE STRING "URL or path containing binary cache files to embed")
 option(MIOPEN_EMBED_BUILD "Build with the set of embed flags." Off)
-option(MIOPEN_USE_MLIR "Use MLIR -- EXPERIMENTAL" Off)
+option(MIOPEN_USE_MLIR "Use MLIR compilation backend" NOT ${MIOPEN_EMBED_BUILD})
```

Comment:
Does CMake support usage of expressions for initialization of options?

Reply:
Honestly speaking I don't know. I was assuming that since the line below works, that this line should also work. But agree on a second look this isn't supported, as from [here](https://cmake.org/cmake/help/latest/command/option.html).

Instead, which do you think is better:
 - Make MIOPEN_USE_MLIR default to be ON
 - Initialize MIOPEN_USE_MLIR not here, but in between line 140 - 148 and turn it on conditionally based on whether it is embed build

Reply:
Please look how I compute and use the default for MIOPEN_USE_COMGR: https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1253/files#r745142441

Reply:
That's indeed a better way to handle it, thanks for informing!